src/providers/dataProvider/dataprovider.test.ts

```diff
@@ -0,0 +1,108 @@
+import { type DataProvider } from 'react-admin'
+import { getDataProvider } from '.'
+import { R_BATCHES } from '../../constants'
+import { DateTime } from 'luxon'
+
+const year: number = 2025
+
+interface NewBatch extends Omit<Batch, 'id'> {
+  readonly id?: number
+}
+
+interface Props {
+  name?: string
+  id?: number
+}
+
+const generateDummyBatchForTesting = ({ name, id }: Props = {}): NewBatch => {
+  return {
+    ...(id !== undefined ? { id } : null),
+    name: name ?? 'Dummy-Batch',
+    startDate: DateTime.now().toFormat('yyyy-MM-dd'),
+    endDate: DateTime.now().plus({ day: 1 }).toFormat('yyyy-MM-dd'),
+    projectCode: '1',
+    batchNumber: `V01/${year}`,
+    yearOfReceipt: String(year),
+    department: 1,
+    project: 1,
+    platform: 1,
+    organisation: 1,
+    maximumProtectiveMarking: 1,
+    remarks: 'remarks-1',
+    receiptNotes: 'receipt-notes-1',
+    createdAt: DateTime.now().toFormat('yyyy-MM-dd'),
+    createdBy: 1
+  }
+}
+
+describe('CRUD operations on each resource', () => {
+  let provider: DataProvider
+  let id: number
+  let batch: Batch
```

Comment:
@TahaKhanAbdalli - I'd like all of these tests to be independent.  I think we can _guarantee_ that by re-creating the data-provider in the `beforeEach`, rather than the `beforeAll`.